schema/snapcraft.json

```diff
@@ -0,0 +1,792 @@
+                            "default": []
+                        },
+                        "organize": {
+                            "type": "object",
+                            "default": {},
+                            "additionalProperties": {
+                                "type": "string",
+                                "minLength": 1
+                            }
+                        },
+                        "filesets": {
+                            "type": "object",
+                            "default": {},
+                            "additionalProperties": {
+                                "type": "array",
+                                "minitems": 1
+                            }
+                        },
+                        "stage": {
+                            "type": "array",
+                            "minitems": 1,
+                            "uniqueItems": true,
+                            "items": {
+                                "type": "string"
+                            },
+                            "default": [
+                                "*"
+                            ]
+                        },
+                        "prime": {
+                            "type": "array",
+                            "minitems": 1,
+                            "uniqueItems": true,
+                            "items": {
+                                "type": "string"
+                            },
+                            "default": [
+                                "*"
+                            ]
+                        },
+                        "override-pull": {
+                            "type": "string",
+                            "default": "snapcraftctl pull"
+                        },
+                        "override-build": {
+                            "type": "string",
+                            "default": "snapcraftctl build"
+                        },
+                        "override-stage": {
+                            "type": "string",
+                            "default": "snapcraftctl stage"
+                        },
+                        "override-prime": {
+                            "type": "string",
+                            "default": "snapcraftctl prime"
+                        },
+                        "parse-info": {
+                            "type": "array",
+                            "minitems": 1,
+                            "uniqueItems": true,
+                            "items": {
+                                "type": "string"
+                            },
+                            "default": []
+                        }
+                    }
+                }
+            }
+        },
+        "plugs": {
+            "type": "object"
+        },
+        "slots": {
+            "type": "object"
+        }
+    },
+    "required": [
+        "name",
+        "parts"
+    ],
+    "anyOf": [
```

Comment:
I don't know about you guys, but I'm gonna forget what some of these things mean if you remove all the comments from the YAML.

Reply:
That's a result of the automated conversion methinks. I'll need to figure out where the old comments need to go to reinstate them...

Reply:
:-(

![image](https://user-images.githubusercontent.com/147548/52290441-fcbb1480-2967-11e9-8e0c-57c6feef73c7.png)

"Comments are not permitted in JSON"

Reply:
Yikes. Why don't we just add code to generate a JSON schema from the YAML then and dump it into e.g. `~/.cache/snapcraft` somewhere, and vscode can pick it up from there? Otherwise we make the most hard-to-understand part of the snapcraft CLI even harder.

Reply:
Or generate the JSON from the YAML when creating the snap, etc.

Reply:
We do not care about the JSON in the snap, we care about it being online.
In the future, we will rev it too.

json does not allow comments, but it could be its own field if jsonschema allows it.

Reply:
I've added the comments using `$comment`. This is probably resolved by that commit 6320560?